module/zfs/spa_stats.c

```diff
@@ -303,6 +304,105 @@ spa_txg_history_init(spa_t *spa)
 	    offsetof(spa_txg_history_t, sth_node));
 }
 
+typedef struct {
+    spa_t *spa;
+    int metaclass;
+} freespace_histogram_stat_t;
+
+static void *
+spa_frag_addr(kstat_t *ksp, loff_t n)
+{
+	if (n == 0) {
+		freespace_histogram_stat_t *stat = ksp->ks_private;
+		switch (stat->metaclass) {
+		case 0:
+			return (spa_normal_class(stat->spa));
+		case 1:
+			return (spa_log_class(stat->spa));
+		case 2:
+			return (spa_embedded_log_class(stat->spa));
+		case 3:
+			return (spa_special_class(stat->spa));
```

Comment:
A 6th metaslab class called `spa_special_embedded_log_class` was added by be1e991a1a83f50c84608709c63d20740fae762f.  Please add this one in too.